internal/osutils/shortcut/shortcut_linux.go

```diff
@@ -0,0 +1,116 @@
+package shortcut
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+)
+
+type Shortcut struct {
+	target string
+	path   string
+}
+
+func New(target, path string) (*Shortcut, error) {
+	if !fileutils.FileExists(target) {
+		return nil, errs.New("Target does not exist")
+	}
+
+	scut := Shortcut{
+		target: target,
+		path:   path,
+	}
+
+	return &scut, nil
+}
+
+type ShortcutSaveOpts struct {
+	GenericName string
+	Comment     string
+	Keywords    string
+	IconData    []byte
+	IconPath    string
+}
+
+func (s *Shortcut) Save(name string, opts ShortcutSaveOpts) error {
```

Comment:
the target and save path go to `New()`, a number of opts go to `ShortcutSaveOpts` and then the Name goes to `Save()`. Frankly I don't follow why these parameters are being cut up this way. Pick one?

Additionally I don't want this API to be completely different depending on what OS you're on. Look at the Windows version and align it with that, or update both. But keep things simple. We don't need three different ways of passing parameters.